src/sources/redis/mod.rs

```diff
@@ -226,10 +246,28 @@ async fn handle_line(
 
                 let events = events.into_iter().map(|mut event| {
                     if let Event::Log(ref mut log) = event {
-                        log.try_insert(log_schema().source_type_key(), Bytes::from("redis"));
-                        log.try_insert(log_schema().timestamp_key(), now);
-                        if let Some(redis_key) = redis_key {
-                            event.as_mut_log().insert(redis_key, key);
+                        log_namespace.insert_vector_metadata(
+                            log,
+                            log_schema().source_type_key(),
+                            "source_type",
+                            Bytes::from("redis"),
+                        );
+                        log_namespace.insert_vector_metadata(
+                            log,
+                            log_schema().timestamp_key(),
+                            "timestamp",
+                            now,
+                        );
+
+                        match log_namespace {
+                            LogNamespace::Vector => log_namespace
+                                .insert_source_metadata("redis", log, "key", "key", key),
+                            LogNamespace::Legacy => {
+                                if let Some(redis_key) = redis_key {
+                                    log_namespace
+                                        .insert_source_metadata("redis", log, redis_key, "key", key)
+                                }
```

Comment:
This is probably intentional but just wondering about the case where `redis_key` is `None` , we don't want to / need to do anything?